specs/governance/alligator.md

```diff
@@ -0,0 +1,390 @@
+and checkpoint data from the token contract to its own state. After copying the data, the `Alligator` MUST update
+the `migrated` mapping to reflect that the address has been migrated.
+
+Before updating the subdelegation, the `subdelegateFromToken` function MUST check the validity of the subdelegation rule.
+
+When updating the subdelegation, the `subdelegateFromToken` function MUST override any previous subdelegation of the
+`_delegator` to the `_delegatee`. Afterwards, this function MUST emit a `Subdelegation` event with the given function
+parameters.
+
+#### `subdelegateBatched`
+
+Allows batch subdelegation of token voting power to multiple addresses with specified subdelegation rules. This
+function is intended to be called by users.
+
+```solidity
+function subdelegateBatched(address[] calldata _delegatees, SubdelegationRules[] calldata _rules) external
+```
+
+This function MUST check if the `msg.sender` has been migrated by checking the `migrated` mapping from its [storage](#storage).
+If it has not been migrated, the `Alligator` MUST copy the delegation and checkpoint data from the token contract
+to its own state. After copying the data, the `Alligator` MUST update the `migrated` mapping to reflect that the address
+has been migrated.
+
+This function MUST check that the length of `_delegatees` and `_rules` are equal. If the lengths are not equal, it MUST
+revert with an error.
+
+The `subdelegateBatched` function MUST iterate over each pair of `_delegatees` address and `_rules` subdelegation rule.
+At every iteration, the function MUST check the validity of the subdelegation rule, and migrate the  delegatee address
+if it has not been migrated. The function MUST then update the subdelegation.
+
+Afterwards, the `subdelegateBatched` function MUST emit a `Subdelegation` event with the given function parameters.
+
+#### `afterTokenTransfer`
+
+Updates the voting power of two addresses (`_from` and `_to`) after a token transfer. This function MUST only be callable
+by the `GovernanceToken` contract.
+
+```solidity
+function afterTokenTransfer(address _from, address _to, uint256 _amount) external
+```
+
+The `Alligator` MUST check if the `_from` or `_to` addresses have been migrated by checking the `migrated` mapping
+from its [storage](#storage). If either address has not been migrated, the `Alligator` MUST copy the delegation
+and checkpoint data from the token contract to its own state. After copying the data, the `Alligator` MUST update
+the `migrated` mapping to reflect that the address has been migrated.
+
+The `afterTokenTransfer` function MUST update the voting power of the `_from` and `_to` addresses with the same logic
+of the `GovernanceToken`.
+
+#### `migrateAccount`
+
+Migrates the delegation state of a given account from the `GovernanceToken` to the `Alligator` contract. This function
+MUST only migrate the `_account` if it has not been migrated yet.
```

Comment:
Thoughts on having this accept an array of address? Generally we want to keep a small ABI but it would be much easier to pass thru an array rather than using multicall

Reply:
yeah i think it would be more useful, tho don't we will often use this function